### PR TITLE
XWIKI-15305 - Introduced 3 new variables to replace the hardcoded bac…

### PR DESCRIPTION
**Pull request by ppantiru**
XWIKI-15305
- Introduced 3 new variables to replace the hardcoded backgrounds and text color. This will be modifiable from the advance field of the color theme, and will be documented on the 'Flamingo Theme Application' page

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/colorThemeInit.vm [hunk 1]
```diff
@@ -48,7 +48,10 @@
 'notificationInfoColor'            : '#336699',
 'notificationSuccessColor'         : '#008000',
 'notificationWarningColor'         : '#D09000',
-'notificationErrorColor'           : '#CC3333'
+'notificationErrorColor'           : '#CC3333',
+'notificationEmailBg'              : '#f5f5f5',
+'notificationEmailInnerBg'         : '#ffffff',
+'notificationEmailTextColor'       : '#1d1f24'
```
Comment (evalica) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/colorThemeInit.vm]: Why did you need to add new variables and could not reuse some of the existing values? 
for example: @notificationEmailBg could have used the @body-bg for example.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/colorThemeMappingOutput.vm [hunk 2]
```diff
@@ -35,4 +35,7 @@
 #mapping("notificationSuccessColor", "@brand-success")
 #mapping("notificationInfoColor", "@brand-info")
 #mapping("notificationWarningColor", "@brand-warning")
-#mapping("notificationErrorColor", "@brand-danger")
\ No newline at end of file
+#mapping("notificationErrorColor", "@brand-danger")
+#mapping("notificationEmailBg", "@notification-email-bg")
+#mapping("notificationEmailInnerBg", "@notification-email-inner-bg")
+#mapping("notificationEmailTextColor", "@notification-email-text-color")
```
Comment (evalica) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/colorThemeMappingOutput.vm]: Also if we were to add new variables, not sure if we should also add the mapping. The mappings is supposed to be used for the old CT variables, not for the new ones. @gdelhumeau wdyt?

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variables.less [hunk 3]
```diff
@@ -11,3 +11,11 @@
...
 @headings-font-weight:           400; // See: http://jira.xwiki.org/browse/XWIKI-10838
 @nav-tabs-active-link-hover-bg:  @xwiki-page-content-bg;
+
+//== Notifications Email colors
+//
+//##
+
+@notification-email-bg: #F5F5F5;
+@notification-email-inner-bg: #ffffff;
+@notification-email-text-color: #1d1f24;
```
Comment (evalica) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variables.less]: there is a top section, called "// New variables provided by XWiki". Could have been added there. Although that category might be added at the bottom, since it might expand faster.

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/MailTemplate.xml [hunk 4]
```diff
@@ -137,7 +138,7 @@
     &lt;/tr&gt;&lt;/table&gt;
 
     ## Events div
-    &lt;div style="background:white;padding:1em; border-left: 1px solid #eaeaea;border-right: 1px solid #eaeaea;border-bottom: 1px solid #ddd;border-top: 1px solid #f0f0f0;"&gt;
+    &lt;div style="background:$!{theme.notificationEmailInnerBg};padding:1em; border-left: 1px solid #eaeaea;border-right: 1px solid #eaeaea;border-bottom: 1px solid #ddd;border-top: 1px solid #f0f0f0;"&gt;
```
Comment (evalica) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/MailTemplate.xml]: I would have reused some existing variables, if they made sense. 
Also instead of using the old .vm variables, you could have made the SSX to use LESS and use LESS @variables.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/colorThemeInit.vm (hunk 1 above)
Comment (ppantiru) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/colorThemeInit.vm]: Because we're talking about the email here, and it's not always ok to use the the general theme colors, for example if we have a dark themed wiki, we might not want the email to be dark as well.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/colorThemeMappingOutput.vm (hunk 2 above)
Comment (gdelhumeau) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/colorThemeMappingOutput.vm]: Indeed, it does not seem necessary here.
